skimage/filters/rank/tests/test_rank.py

```diff
@@ -59,6 +59,19 @@ def test_subtract_mean_underflow_correction(dtype):
 ref_data_3d = dict(np.load(fetch('data/rank_filters_tests_3d.npz')))
 
 
+@pytest.mark.parametrize(
+    'func', ['autolevel', 'equalize', 'gradient', 'maximum',
+            'mean', 'geometric_mean', 'geometric_mean', 'subtract_mean',
+            'median', 'minimum', 'modal', 'enhance_contrast', 'pop',
+            'sum', 'threshold', 'noise_filter', 'entropy', 'otsu', 'majority'
+    ])
```

Comment:
Why not reference the functions directly instead of going the detour via `getattr(rank, func)`? Referencing them directly would make live easier for refactoring tools and IDEs.

Also you've got `geometric_mean` mean in there twice. Could you check the list again?

Reply:
You're right, I'll remove the extra function 😅

About the `getattr`, my issue was with importing all of these functions in the test.
I didn't want to use a wildcard (*) because it conflicts with:
```python
from skimage.filters.rank import __all__ as all_rank_filters
```
And specifically importing all of these functions seemed redundant.
I saw the `getattr` approach in another test, and it seemed to solve this problem.

Let me know what you think.

Reply:
Yeah, not using `from ... import *` is a good instinct. Maybe I'm misunderstanding you, but you don't need to import every function explicitly. How about
```python


@pytest.mark.parametrize(
    'func', [rank.autolevel, rank.equalize, ... rank.majority
    ])
def test_1d_input_raises_error(func):
    image = np.arange(10)
    footprint = disk(3)
    with pytest.raises(ValueError,
            match='input image should be a 2-dimensional or 3-dimensional array;'):
        func(image, footprint)
```
I'd just copy the list of strings you already have into a Python console and print `rank.autolevel, ...` part from that.